internal/app/mapper/mapper.go

```diff
@@ -74,7 +74,27 @@ func isMatch(matchPredicate *matchPredicate, typeURL string, node *core.Node) (b
 		return false, err
 	}
 
-	return isNodeMatch || isRequestTypeMatch(matchPredicate, typeURL) || isAnyMatch(matchPredicate), nil
+	if isNodeMatch {
```

Comment:
@jessicayuen I know this function has seen the most style comments. Any ideas how to handle this mess now?

Reply:
Working with protos is a bit clunky, but can we find out which type we're going to handle and have a switch instead of this sequence of if-else blocks?

Reply:
@eapolinario can you tell bit more about `which type we're going to handle` ?

Reply:
@jyotimahapatra I think this is ok. I can't see us getting around this if we want to return early on result match or error scenarios.